cmd/s3-gw/app.go

```diff
@@ -189,31 +197,45 @@ func (a *App) initMetrics() {
 	a.metrics = newAppMetrics(a.log, gateMetricsProvider, a.cfg.GetBool(cfgPrometheusEnabled))
 }
 
-func (a *App) initResolver() {
-	var err error
-	a.bucketResolver, err = resolver.NewBucketResolver(a.getResolverConfig())
+func (a *App) initResolver(ctx context.Context) {
```

Comment:
I left func `initResolver` here, but reworked the common resolvers constructor. Depending on `endpoint` it will return either NoOpResolver or NNSResolver